### PR TITLE
CIP-0084? | Ledger evolution (meta)

### PR DESCRIPTION
**Pull request by JaredCorduan**
This CIP aims to make the `Cardano ledger` a registered category of the CIP process.

---

([rendered proposal in branch](https://github.com/JaredCorduan/CIPs/blob/ledger-evolution/CIP-%3F%3F%3F%3F/README.md))

### CODE REVIEW

File: CIP-????/README.md
Comment (michaelpj) [inline CIP-????/README.md:161]: I think this isn't needed any more since there's a category that does the job of aggregating things.

File: CIP-????/README.md
Comment (michaelpj) [inline CIP-????/README.md:193]: I still think ledger events should pretty clearly be "out". They don't affect the operation of the Cardano network. A competing implementation could omit them and be perfectly correct.

**Review by michaelpj — APPROVED** [review]

File: CIP-????/README.md
Comment (rphair) [inline CIP-????/README.md:135]: This section could also use a brief reference like what @michaelpj put in CIP-0035 about categorising CIPs which authors might be tempted to categorise as both `plutus` and `ledger`: https://github.com/cardano-foundation/CIPs/tree/master/CIP-0035#scope

The pointer to CIP-00035 will work for many users, but some authors & third parties will be helped by quoting or summarising this paragraph from it:
> Additionally, there is significant overlap with the Ledger category around the ledger-script interface and the protocol parameters. CIPs which change these parts of Cardano should generally use the Plutus category and not the Ledger category, although the Editors may ask the Ledger reviewers to comment.

File: CIP-????/README.md
Comment (rphair) [inline CIP-????/README.md:192]: @JaredCorduan maybe you were planning to add the required _Path to Active_ here & this is still a draft?  If not prepared yet, this could again resemble CIP-0035 with "Ledger" substituted for "Plutus" ... i.e. it's accepted when the Ledger team accepts the CIP, and implemented when the Ledger team implements it ... or something more articulated if that would suit you better 😎

https://github.com/cardano-foundation/CIPs/tree/master/CIP-0035#path-to-active

Comment (rphair) [inline CIP-????/README.md:192]: ... and another obligatory section: your chose software licensing model and/or link, as per https://github.com/cardano-foundation/CIPs/tree/master/CIP-0001#licensing

**Review by rphair — COMMENTED** [review]
This looks like a definitive statement from the Ledger team so I'd be ready to approve if these issues are addressed, also pending any further input from @michaelpj & others 😎

File: CIP-????/README.md
Comment (JaredCorduan) [reply CIP-????/README.md:135]: great idea, thanks!

File: CIP-????/README.md
Comment (JaredCorduan) [reply CIP-????/README.md:211]: great, I'm happy to keep consistent wording with CIP-35.

File: CIP-????/README.md
Comment (JaredCorduan) [inline CIP-????/README.md:141]: I see that CIP-35 has a [Reviewers](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0035#the-plutus-reviewers) section now. I hadn't planned on adding a table like that, I suspect that the folks I would put in such a table would end up review ledger CIPs anyway. Should I re-consider? cc @michaelpj

**Review by rphair — APPROVED** [review]
All the details that I marked have been filled in & elaborated, so I'm happy to approve this while keeping an eye on further review.

File: CIP-????/README.md
Comment (michaelpj) [reply CIP-????/README.md:141]: I'm not sure. I put it there because the new CIP-001 somewhat implicitly has the concept of project maintainers/reviewers who are the people who get called upon to review CIPs in that category, and I thought it was better to make it explicit who that was.

File: CIP-????/README.md
Comment (KtorZ) [inline CIP-????/README.md:2]:
```suggestion
CIP: 84
```

File: CIP-????/README.md
Comment (KtorZ) [inline CIP-????/README.md:6]:
```suggestion
Authors: 
  - Jared Corduan <jared.corduan@iohk.io>
```

File: CIP-????/README.md
Comment (KtorZ) [inline CIP-????/README.md:4]:
```suggestion
Status: Active
```

File: CIP-????/README.md
Comment (KtorZ) [inline CIP-????/README.md:8]:
```suggestion
Created: 2023-01-30
Discussions: 
  - https://github.com/cardano-foundation/CIPs/pull/456
```

File: CIP-????/README.md
Comment (KtorZ) [inline CIP-????/README.md:12]:
```suggestion
```

File: CIP-????/README.md
Comment (KtorZ) [inline CIP-????/README.md:18]:
```suggestion
## Motivation: why is this CIP necessary?
```

File: CIP-????/README.md
Comment (KtorZ) [inline CIP-????/README.md:38]:
```suggestion
### Specification
```

File: CIP-????/README.md
Comment (KtorZ) [inline CIP-????/README.md:141]:
```suggestion
```

File: CIP-????/README.md
Comment (KtorZ) [inline CIP-????/README.md:164]: I think this section would benefit from a little bit of rework to state more clearly what is a potential candidate for a `Ledger` CIP, and what isn't. I'd suggest to make two lists with bullet points. 

I would be keen on seeing a mention of the RSS here (in either of the list).

cc @JaredCorduan

File: CIP-????/README.md
Comment (KtorZ) [inline CIP-????/README.md:181]:
```suggestion
| Jared Corduan | jared.corduan@iohk.io | @JaredCorduan  |
```

File: CIP-????/README.md
Comment (KtorZ) [inline CIP-????/README.md:182]:
```suggestion
| Andre Knispel | andre.knispel@iohk.io | @WhatisRT        |
```

File: CIP-????/README.md
Comment (KtorZ) [inline CIP-????/README.md:183]: I think the "Specification" would benefit from a short section about what it takes for a ledger CIP to _become active_. Are you expecting people to submit PRs to the ledger, or would you rather avoid people doing so (and have only the core maintainers do the work that's agreed on CIPs)?

File: CIP-????/README.md
Comment (KtorZ) [inline CIP-????/README.md:184]:
```suggestion
## Rationale: how does this CIP achieve its goals?
```

File: CIP-????/README.md
Comment (JaredCorduan) [reply CIP-????/README.md:164]: @KtorZ do you mean that the content is fine, but that it would help to use bullets to make it more visually clear?

What RSS are you referring to?

Comment (KtorZ) [reply CIP-????/README.md:165]: > @JaredCorduan do you mean that the content is fine, but that it would help to use bullets to make it more visually clear?

Yes, the suggestion is purely _structural_. From the perspective of a contributor, I'd like to easily grasp in one look what are suitable topics for a ledger CIP and what aren't.

> What RSS are you referring to?

I am referring to anything that touches the rewards and incentives schemes. This is a common topic of discussion in CIPs; ranging from suggestions of new values for protocol parameters to changes in the reward equations (in particular w.r.t to the pledge influence) or complete redesign of the incentives. 

Those CIPs have historically gotten very little attention and are hard to move forward. I'd appreciate clarity as to whether the `Ledger` category now cover them, or not. If the intent is to cover this area, then I'd also love the process to be specified in more detail in this proposal. So far, it seems that RSS questions have required the involvement of IO research but the process hasn't always / ever been smooth as there's little availability from the research side. 

Plus, all CIPs in this situation never got a clear answer from IO / the ledger team regarding their "acceptance". That's why from an edition perspective, we've recently decided to simply mark all of them as _Inactive_ because we don't have any procedure by which we can propose them and move them forward. 

If this proposal still doesn't address the case of RSS proposals, then let's state it clearly here too.

Comment (JaredCorduan) [reply CIP-????/README.md:165]: > I'd like to easily grasp in one look what are suitable topics for a ledger CIP and what aren't.

I definitely appreciate your help to make this more readable and friendly. Really there is just one rule:

```
every implementation of the Cardano ledger needs to be standardized on the details.
```

The paragraphs below this, that I think you are suggesting making into a bullet list, are just elaborations on this and exceptions. Maybe I could bold the main criterion sentence? Would adding bullets to the other things make the relevance of the main criterion seem less apparent?

Comment (JaredCorduan) [reply CIP-????/README.md:165]: regarding the rewards and incentives schemes, my hope was that it would be clear that indeed they are in scope:

```
every implementation of the Cardano ledger needs to be standardized on the details.
```

The caveat is listed below:

```
 For example, changes to the incentives model (such as changing the reward calculation)
  should include a proper game theoretic analysis.
```

Of course, as you can see, I've given no guidance on what is "proper" or who decides. 

> Those CIPs have historically gotten very little attention and are hard to move forward

I think it is more that there are only so many things we can do at once, and we have been focused on babbage and voltaire.

File: CIP-????/README.md
Comment (JaredCorduan) [reply CIP-????/README.md:184]: > Are you expecting people to submit PRs to the ledger, or would you rather avoid people doing so (and have only the core maintainers do the work that's agreed on CIPs)?

Something in the middle, I think. This involves perhaps a big topic that we should discuss.

Features do not get implemented in isolation in the ledger. We collect up a bunch of features, see how they interact, design a specification for a new ledger era, and then implement the era. I actually wrote up an earlier draft of this CIP that was much different than what I've submitted, where "ledger era" was its own ledger sub-type. In the end, I talked myself out of this since it could be argued that this is more of the release process. The ledger team would of course love help with PRs, but it would need to match our existing workflow, which is to build out the code once the era is settled. I'm not sure how the bundling should be involved in the CIP process.

Comment (JaredCorduan) [reply CIP-????/README.md:184]: > short section about what it takes for a ledger CIP to become active

I could probably use some guidance from the CIP committee as to how to organize this. These are the stages in my mind:

* everyone agrees it is a good change
* it is selected for inclusion in some ledger era
* it is added to the formal spec
* it is implemented
* the hard fork happens and it is live

File: CIP-????/README.md
Comment (rphair) [reply CIP-????/README.md:165]: @JaredCorduan yes I think it does appear that the need to categorise RSS proposals _somehow_ would be satisfied by giving them a `Ledger` category... while the comment you quote directly above also indicates that the consideration of these proposals will generally depend on expertise & review outside the Ledger team (i.e. a "disclaimer" so that their responsibility would not fall exclusively upon them).

So in my opinion this question has no reason to block your proposal as it currently stands... but I'll leave this conversation open so we can see it in our CIP meeting (1 hour from now). (cc @KtorZ)

**Comment by rphair** [comment]
@JaredCorduan this is up for Last Check at our CIP meeting today & I'm going to recommend that this be merged... one issue perhaps to discuss at the meeting is if the text still requires any specific mention of the RSS proposals, other than your suggested language to cover them, which I think is currently adequate (https://github.com/cardano-foundation/CIPs/pull/456#discussion_r1135174031).

**Comment by rphair** [comment]
OK counter to my last comments, from meeting (@KtorZ) it seems that, although clarification is still needed, it should be literal clarification that RSS proposals are _out_ of scope for Ledger & that proposals categorised as such should be straightforwardly applicable to the Leger.

**Comment by rphair** [comment]
@JaredCorduan that was also my understanding & the basis of my last couple of comments, but I had to retract it after today's meeting discussion suggested that the `Ledger` category should only be applied to the more traditionally ledger-oriented things (plus governance now obviously).  We don't have a recording of the meeting yet so I'll leave it to @KtorZ to qualify that statement.

**Review by KtorZ — APPROVED** [review]
Happy to move this one forward, modulo one concern:

- In the following sentence: "Some changes may require getting an expert opinion." Who qualifies as "an expert" here? Is it more of a general notion as in, someone that has invested enough efforts in studying those topics or are we expecting people with specific credentials (e.g. PhD in game theory or similar).

**Comment by JaredCorduan** [comment]
> * Who qualifies as "an expert" here? Is it more of a general notion as in, someone that has invested enough efforts in studying those topics or are we expecting people with specific credentials (e.g. PhD in game theory or similar).

This is a really great question, one that I have struggled with myself and failed to find a good answer. I definitely do not think that credentials are the answer. The best I can come up with right now is to let the CIP editors and/or the ledger category-specific reviewers make judgements calls.

**Comment by rphair** [comment]
> let the CIP editors and/or the ledger category-specific reviewers make judgements calls.

@KtorZ @JaredCorduan this criteria is fine with me.  Since we can't count on connections to whatever few definable experts there may be, we are about to allow (and did allow in 2021) any RSS proposal to be accepted as long as peer + Ledger team review agree that it establishes rationally why it could be desirable.  There is no point at which an "expert" can be expected or required by the CIP process, so it would be logical to strike out that whole bullet point.

An alternative would be to write it differently in an a way that is both bureaucratically comfortable _and_ true, like:

- Though proposals can be accepted solely on the basis of peer and Ledger team review, some areas (e.g. changes to the incentives model) might only considered ready for implementation if accompanied by an opinion from an expert designated by the implementor (e.g. with a proper game theoretic analysis).

**Comment by rphair** [comment]
@WhatisRT @lehins (as the remaining ledger reviewers): so this proposal can be merged, would you please respond to https://github.com/cardano-foundation/CIPs/pull/456#issuecomment-1585558784 adding your preferred language regarding RSS proposals... or give your permission to the CIP editors to modify the language to qualify what is meant by an "expert" as suggested above?  cc @JaredCorduan @KtorZ @Ryun1

**Comment by WhatisRT** [comment]
I think having some language in the CIP about expert opinions is good, so I'd rather keep that line & amend it as you've proposed.

File: CIP-????/README.md
Comment (rphair) [inline CIP-????/README.md:174]:
```suggestion
* Though proposals can be accepted solely on the basis of peer and Ledger team review, some areas (e.g. changes to the incentives model) might only considered ready for implementation if accompanied by an opinion from an expert designated by the implementor (e.g. with a proper game theoretic analysis).
```
Note the parallel example (two e.g.'s here, like logical propositions, assumed to relate to each other) was argued for by @JaredCorduan somewhere in the CIP-1694 discussion but I can't find the reference to that comment. 😅

**Review by rphair — COMMENTED** [review]
@WhatisRT would this be acceptable?

File: CIP-????/README.md
Comment (JaredCorduan) [reply CIP-????/README.md:174]: I :heart: the suggestion!

**Pull request merged**
